cmd/minikube/cmd/root.go

```diff
@@ -94,6 +94,20 @@ var RootCmd = &cobra.Command{
 Please use --v=3 to show libmachine logs, and --v=7 for debug level libmachine logs
 `)
 		}
+
+		//TODO(r2d4): config should not reference API
+		clientType = configCmd.GetClientType()
+
+		logDir := pflag.Lookup("log_dir")
+		if !logDir.Changed {
+			logDir.Value.Set(constants.MakeMiniPath("logs"))
+		}
+		cobra.OnInitialize(initConfig)
```

Comment:
OnIntialization adds that function as an initializer which gets called here
https://github.com/spf13/cobra/blob/master/command.go#L593

That code is called before the PersistentPreRun, so it we won't actually run our initializer function.  

This was also originally in init() before a recent merge